src/Fixer/NamespaceNotation/BlankLinesBeforeNamespaceFixer.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\NamespaceNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolverInterface;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use Symfony\Component\OptionsResolver\Options;
+
+/**
+ * @author Graham Campbell <hello@gjcampbell.co.uk>
+ * @author Greg Korba <greg@codito.dev>
+ */
+final class BlankLinesBeforeNamespaceFixer extends AbstractFixer implements WhitespacesAwareFixerInterface, ConfigurableFixerInterface
+{
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Controls blank lines before a namespace declaration.',
+            [
+                new CodeSample("<?php  namespace A {}\n"),
+                new CodeSample("<?php  namespace A {}\n", ['min_line_breaks' => 1]),
+                new CodeSample("<?php\n\n\nnamespace A{}\n", ['max_line_breaks' => 2]),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_NAMESPACE);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run after BlankLineAfterOpeningTagFixer.
+     */
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('min_line_breaks', 'Minimum line breaks that should exist before namespace declaration.'))
```

Comment:
Why `min_line_breaks` and `max_line_breaks`? Wouldn't be more developer/user friendly to have `min_blank_lines` and `max_blank_lines`?

If `min_line_breaks` and `max_line_breaks` are to stay shouldn't the fixer be named `line_breaks_before_namesce`?

Reply:
Reasoning is in the description 😝. In terms of fixer's name: I wanted to keep consistency with current fixers and other fixers related to blank lines.

Reply:
Weird, but fair enough.

I thought that line with `<?php` is reserved for `declare(strict_types=1);`, but we can't expect that strict types will be used.